crates/federated-server/src/server.rs

```diff
@@ -173,17 +179,27 @@ pub async fn router<R: engine::Runtime, SR: ServerRuntime>(
     };
 
     let state = ServerState::new(
-        engine,
+        engine.clone(),
         config.request_body_limit.bytes().max(0) as usize,
         server_runtime.clone(),
     );
 
     let mut router = server_runtime
         .base_router()
         .unwrap_or_default()
-        .route(path, get(engine_execute).post(engine_execute))
+        .route(path, get(graphql_execute).post(graphql_execute))
         .route_service(websocket_path, WebsocketService::new(websocket_sender));
 
+    let ct = match config.mcp {
```

Comment:
Seems like  we should re-use the cancellation token to handle graceful shutdown everywhere in the future, no? Doesn't seem like you'll be gracefully shutting down MCP here. (doesn't matter for this PR)

Reply:
Yeah... it's to stop the spawned SSE servers. These will be gone when the crate implements the new protocol, and we don't need to keep an SSE alive anymore.